patch/llpcConfigBuilderBase.h

```diff
@@ -68,12 +73,25 @@ class ConfigBuilderBase
     void SetLdsSizeByteSize(Util::Abi::HardwareStage hwStage, uint32_t value);
     void SetEsGsLdsSize(uint32_t value);
 
+    void AppendConfig(llvm::ArrayRef<Util::Abi::PalMetadataNoteEntry> config);
+    void AppendConfig(uint32_t key, uint32_t value);
+
+    template<typename T>
+    void AppendConfig(const T& config)
+    {
+        static_assert(T::containsPalAbiMetadataOnly,
+                      "may only be used with structs that are fully metadata notes");
```

Comment:
may -> May
I think we use Uppercase-initial for assert messages like what we have done for normal comments.